.github/workflows/release.yml

```
@@ -38,3 +38,76 @@ jobs:
         if: ${{startsWith(github.ref, 'refs/tags/') }}
         with:
           files: pie.phar
+
+  docker-binary-only-image:
+    needs: build-phar
+    name: Docker binary-only image
+    runs-on: ubuntu-latest
+    if: ${{ startsWith(github.ref, 'refs/tags/') }}
+
+    permissions:
+      # attestations:write is required for build provenance attestation.
+      attestations: write
+      # id-token:write is required for build provenance attestation.
+      id-token: write
+      # packages:write is required to publish Docker images to GitHub's registry.
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Fetch built PHAR from artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: pie-${{ github.sha }}.phar
+
+      - name: Verify the PHAR
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: gh attestation verify pie.phar --repo ${{ github.repository }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          flavor: |
+            latest=false
+          images: ghcr.io/${{ github.repository }}
+          tags: |
+            type=raw,value=bin
+            type=semver,pattern={{version}}-bin
+            type=semver,pattern={{major}}.{{minor}}-bin
+            type=semver,pattern={{major}}-bin
```

Thank you for all the work you're doing on this @Wirone - I think this is nearly close to being ready. At this stage, given the early stage of releases, I would like to reduce these tags down to:

```suggestion
          tags: |
            type=raw,value=bin
            type=semver,pattern={{version}}-bin
```

Once we release `1.0.0` we can re-open a fresh discussion about the `:{{major}}.{{minor}}-bin` and `:{{major}}-bin` tags; but at this stage, this is unnecessary, and since semver prior to `1.0.0` works differently anyway (i.e. each minor may contain a BC break), then this currently may have unintended consequences.

I had considered _only_ keeping `:bin`, but I think keeping `:{{version}}-bin` has the benefit of people being able to pin to a specific version should there be a compatibility issue or bug introduced in a newer release.

---

Done in [460f0c3](https://github.com/php/pie/pull/122/commits/460f0c380159bf8040a45d42b946279dfba64bf6), let me know if the `@TODO` is OK for you, to track this before v1.0.

---

@asgrim FYI the link `https://ghcr.io/php/pie` used in usage docs currently returns 404 because there are no built images in the registry. Please see `https://ghcr.io/wirone/php-pie` (my fork's short URL) to see it redirects to working registry, so it will also work here after first release.